Mathlib/SetTheory/Surreal/Division.lean

```diff
@@ -0,0 +1,714 @@
+    (ih_xr : ∀ j, (x.moveRight j).inv'.Numeric) :
+    (∀ i, (x.inv'.moveLeft i).Numeric) ∧ (∀ j, (x.inv'.moveRight j).Numeric) := by
+  constructor <;> (
+    intro i
+    rcases x with ⟨xl, xr, xL, xR⟩
+    exact invVal_numeric ih_xl ih_xr x_num i
+  )
+
+lemma inv'_numeric_left (ih_xl : ∀ i, 0 < x.moveLeft i → (x.moveLeft i).inv'.Numeric)
+    (ih_xr : ∀ j, (x.moveRight j).inv'.Numeric) :
+    ∀ i, (x.inv'.moveLeft i).Numeric := (inv'_numeric x_num x_pos ih_xl ih_xr).1
+
+lemma inv'_numeric_right (ih_xl : ∀ i, 0 < x.moveLeft i → (x.moveLeft i).inv'.Numeric)
+    (ih_xr : ∀ j, (x.moveRight j).inv'.Numeric) :
+    ∀ j, (x.inv'.moveRight j).Numeric := (inv'_numeric x_num x_pos ih_xl ih_xr).2
+
+/-! ### Helper lemmas for the main proof
+
+  These lemmas do the heavy lifting within the proof of the main result `onag_1_10`.
+  They are stated with additional hypotheses which will be satisfied by induction
+  during the main proof.
+
+-/
+
+/-- Given `y''`, return `(x', x'_inv, y')` -/
+def components {l r} {L : l → PGame} {R : r → PGame}
+    (h3 : ∀ i, 0 < L i → (L i).inv'.Numeric)
+    (h4 : ∀ j, (R j).inv'.Numeric) (h5 : (PGame.mk l r L R).Numeric) {b : Bool}
+    (x_pos : 0 < PGame.mk l r L R)
+    (i'' : InvTy {i // 0 < L i} r b) :=
+  match i'' with
+  | InvTy.zero => (0, 0, 0)
+  | InvTy.left₁ i j =>
+    (mk (R i) (h5.2.2 i), mk (R i).inv' (h4 i),
+      mk ((PGame.mk l r L R).inv'.moveLeft j) (by apply inv'_numeric_left <;> tauto))
+  | InvTy.left₂ i j =>
+    (mk (L i) (h5.2.1 i), mk (L i).inv' (h3 i i.2),
+      mk ((PGame.mk l r L R).inv'.moveRight j) (by apply inv'_numeric_right <;> tauto))
+  | InvTy.right₁ i j =>
+    (mk (L i) (h5.2.1 i), mk (L i).inv' (h3 i i.2),
+      mk ((PGame.mk l r L R).inv'.moveLeft j) (by apply inv'_numeric_left <;> tauto))
+  | InvTy.right₂ i j =>
+    (mk (R i) (h5.2.2 i), mk (R i).inv' (h4 i),
+      mk ((PGame.mk l r L R).inv'.moveRight j) (by apply inv'_numeric_right <;> tauto))
+
+/-- The identity `1 - x*y'' = (1 - x * y')*(x' - x)/x'` -/
+lemma eq1 {l r} {L : l → PGame} {R : r → PGame}
+    (h3 : ∀ i, 0 < L i → (L i).inv'.Numeric)
+    (h4 : ∀ j, (R j).inv'.Numeric) (h5 : (PGame.mk l r L R).Numeric) {b : Bool}
+    (x_pos : 0 < PGame.mk l r L R)
+    (inv_l : ∀ (i : { i // 0 < L i }), mk (L i) (h5.2.1 i) * mk (L i).inv' (h3 i i.2) = 1)
+    (inv_r : ∀ j, mk (R j) (h5.2.2 j) * mk (R j).inv' (h4 j) = 1)
+    (i'' : InvTy {i // 0 < L i} r b) :
+    let (x', x'_inv, y') := components h3 h4 h5 x_pos i''
+    let x := mk (PGame.mk l r L R) h5
+    let y'' := match b with
+      | false => mk ((PGame.mk l r L R).inv'.moveLeft i'')
+        (by apply inv'_numeric_left <;> tauto)
+      | true => mk ((PGame.mk l r L R).inv'.moveRight i'')
+        (by apply inv'_numeric_right <;> tauto)
+    match i'' with
+    | InvTy.zero => true
+    | _ => 1 - x*y'' = (1 - x * y')*(x' - x)*x'_inv := by
+  cases i'' <;> simp only [components]
+  all_goals
```

Comment:
If I understand correctly, what this inductive proof does is
a) take care of all numericity hypotheses first
b) cast your values into elements of `Surreal*`
c) delegate all the actual calculations to `field_simp; ring`
Can you add a few short comments in the proof explaining this?